File: modules/agent/src/models/TenantPreferences.py
from sqlalchemy import Column, Text, DateTime, BigInteger, ForeignKey, func
from sqlalchemy.orm import relationship
from models import Base


class TenantPreferences(Base):
    """TenantPreferences SQLAlchemy model (theme settings for entire tenant)."""

    __tablename__ = "Tenant_Preferences"

    id = Column(BigInteger, primary_key=True, autoincrement=True)
    tenant_id = Column(BigInteger, ForeignKey("Tenant.id", ondelete="CASCADE"), nullable=False, unique=True, index=True)
    theme = Column(Text, nullable=False, default="light")
    created_at = Column(DateTime(timezone=True), server_default=func.now(), nullable=False)
    updated_at = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now(), nullable=False)

    # Relationships
    tenant = relationship("Tenant", back_populates="preferences", uselist=False)

File: modules/agent/internal/models/TenantPreferences.go
package models

import "time"

type TenantPreferences struct {
	ID        int64     `gorm:"primaryKey;autoIncrement" json:"id"`
	TenantID  int64     `gorm:"uniqueIndex;not null" json:"tenant_id"`
	Theme              string `gorm:"not null;default:light" json:"theme"`
	MinCrawlerInterval int    `gorm:"not null;default:10" json:"min_crawler_interval"`
	CreatedAt time.Time `gorm:"autoCreateTime" json:"created_at"`
	UpdatedAt time.Time `gorm:"autoUpdateTime" json:"updated_at"`
}

func (TenantPreferences) TableName() string {
	return "Tenant_Preferences"
}
